arcgis-ios-sdk-samples/Route and directions/Navigate route/NavigateRouteViewController.swift

```diff
@@ -159,6 +152,36 @@ class NavigateRouteViewController: UIViewController {
         }
     }
     
+    /// Set route tracker, data source and location display with a solved route result.
+    ///
+    /// - Parameter routeResult: An `AGSRouteResult` object.
+    func setNavigation(with routeResult: AGSRouteResult) {
+        // Set the route tracker
+        routeTracker = makeRouteTracker(result: routeResult)
+        
+        // Set the mock location data source.
+        let firstRoute = routeResult.routes.first!
+        directionsList = firstRoute.directionManeuvers
+        let mockDataSource = makeDataSource(route: firstRoute)
+        initialLocation = mockDataSource.locations?.first
+        // Create a route tracker location data source to snap the location display to the route.
+        let routeTrackerLocationDataSource = AGSRouteTrackerLocationDataSource(routeTracker: routeTracker, locationDataSource: mockDataSource)
+        
+        // Set location display.
+        mapView.locationDisplay.dataSource = routeTrackerLocationDataSource
+        mapView.locationDisplay.autoPanMode = .navigation
+        // If the user navigates the map view away from the location display, activate the recenter button.
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in
+            self?.recenterBarButtonItem.isEnabled = true
+            self?.mapView.locationDisplay.autoPanModeChangedHandler = nil
+        }
```

Comment:
Could this be replaced by a call to `recenter()`? It seems like that method is doing most of this already and it would be good not to do the same thing in multiple places.